extensions/objects/templates/filepath-editor.directive.ts

```diff
@@ -762,7 +774,40 @@ angular.module('oppia').directive('filepathEditor', [
               AlertsService.addWarning('Could not get resampled file.');
               return;
             }
-            ctrl.postImageToServer(dimensions, resampledFile);
+            ctrl.saveImage(dimensions, resampledFile);
+          }
+        };
+
+        ctrl.saveImageToLocalStorage = function(
+            dimensions, resampledFile, imageType) {
+          var filename = ImageUploadHelperService.generateImageFilename(
+            dimensions.height, dimensions.width, imageType);
+          var reader = new FileReader();
+          reader.onload = function() {
+            var imageData = reader.result;
+            ImageLocalStorageService.saveImage(filename, imageData);
+            var img = new Image();
+            img.onload = function() {
+              ctrl.setSavedImageFilename(filename, true);
+              var dimensions = (
+                ImagePreloaderService.getDimensionsOfImage(filename));
+              ctrl.imageContainerStyle = {
+                height: dimensions.height + 'px',
+                width: dimensions.width + 'px'
+              };
+              $scope.$apply();
+            };
+            img.src = getTrustedResourceUrlForImageFileName(filename);
+          };
+          reader.readAsDataURL(resampledFile);
+        };
+
+        ctrl.saveImage = function(
+            dimensions, resampledFile, imageType = 'png') {
+          if (ContextService.areImagesSavedInLocalStorage()) {
+            ctrl.saveImageToLocalStorage(dimensions, resampledFile, imageType);
+          } else {
+            ctrl.postImageToServer(dimensions, resampledFile, imageType);
```

Comment:
Should this method delete the images from localStorage (if they exist)?

Reply:
This is not the function that sends the images in local storage to server, this is the function that is called in normal operation when an image is directly sent to the server.
The images are sent to the server from localStorage along with the POST request that creates the corresponding entity (according to the design doc).

Reply:
Oh got it. Thanks!